ansible_collections/arista/avd/plugins/test/defined.py

```diff
@@ -69,48 +69,55 @@ def defined(value, test_value=None, var_type=None, fail_action=None, var_name=No
     boolean
         True if variable matches criteria, False in other cases.
     """
+    display = Display()
     if isinstance(value, Undefined) or value is None:
         # Invalid value - return false
         if str(fail_action).lower() == 'warning':
+            display._warns = {}
             if var_name is not None:
-                Display().warning(f"{var_name} was expected but not set. Output may be incorrect or incomplete!")
+                display.warning(f"{var_name} was expected but not set. Output may be incorrect or incomplete!")
             else:
-                Display().warning("A variable was expected but not set. Output may be incorrect or incomplete!")
+                display.warning("A variable was expected but not set. Output may be incorrect or incomplete!")
         elif str(fail_action).lower() == 'error':
             if var_name is not None:
                 raise AnsibleError(f"{var_name} was expected but not set!")
             else:
                 raise AnsibleError("A variable was expected but not set!")
+        if run_tests:
+            return False, display._warns
         return False
+
     elif test_value is not None and value != test_value:
         # Valid value but not matching the optional argument
         if str(fail_action).lower() == 'warning':
+            display._warns = {}
             if var_name is not None:
-                Display().warning(f"{var_name} was set to {value} but we expected {test_value}. Output may be incorrect or incomplete!")
+                display.warning(f"{var_name} was set to {value} but we expected {test_value}. Output may be incorrect or incomplete!")
             else:
-                Display().warning(f"A variable was set to {value} but we expected {test_value}. Output may be incorrect or incomplete!")
+                display.warning(f"A variable was set to {value} but we expected {test_value}. Output may be incorrect or incomplete!")
         elif str(fail_action).lower() == 'error':
             if var_name is not None:
                 raise AnsibleError(f"{var_name} was set to {value} but we expected {test_value}!")
             else:
                 raise AnsibleError(f"A variable was set to {value} but we expected {test_value}!")
+        if run_tests:
+            return False, display._warns
         return False
     elif str(var_type).lower() in ['float', 'int', 'str', 'list', 'dict', 'tuple', 'bool'] and str(var_type).lower() != type(value).__name__:
```

Comment:
I think there is a bug in this line. If var_type == ```abc``` this line fails and goes to else, which return True. Any comments!

Reply:
I am trying ```elif str(var_type).lower() is not None``` and then test the above condition

Reply:
since this is the last test, to test the var_type, it is safe to assume that everything else passed, hence if the var_type is not one of the supported classes we just skip this test.

Reply:
Thanks Tony , Updated the test and tested it